yjit/src/core.rs

```diff
@@ -1425,13 +2008,17 @@ pub fn take_version_list(blockid: BlockId) -> VersionList {
 fn get_num_versions(blockid: BlockId, inlined: bool) -> usize {
     let insn_idx = blockid.idx.as_usize();
     match get_iseq_payload(blockid.iseq) {
+
+        // FIXME: this counting logic is going to be expensive.
+        // We should avoid it if possible
```

Comment:
like always encoding `SetInlineBlock` as a first instruction? 🤔

Reply:
not sure, but it seemed expensive-ish to decode all the contexts to count this?

Reply:
right, I just raised a potential idea. feel free to leave it in this PR.